api/src/legacy_soap_api/grantors/services/get_submission_list_expanded_response.py

```diff
@@ -62,7 +76,8 @@ def transform_submission(submission: ApplicationSubmission) -> dict[str, str | d
 def get_submission_list_expanded_response(
     db_session: db.Session,
     get_submission_list_expanded_request: schemas.GetSubmissionListExpandedRequest,
-) -> schemas.GetSubmissionListExpandedResponse:
+    soap_request: SOAPRequest,
+) -> schemas.GetSubmissionListExpandedResponse | list:
```

Comment:
In what case would we return something other than the response type? If there are no results like the status case, wouldn't we still return a response object, but the submission info list would be empty?

The idea I suggested in a prior PR was to make a new function that handles the DB query logic and returns the results (and can shortcut to return `[]`) and this logic just uses that list.


This function should just be:
```py
submissions = get_submissions( ... )

for submission in submissions:
    submission_list_obj = transform_submission(submission)
    info.append(schemas.SubmissionInfo(**submission_list_obj))
return schemas.GetSubmissionListExpandedResponse(
    success=True, available_application_number=len(info), submission_info=info
    )
```